BlockManager copy constructor fix

tshmilnvidia opened the pull request:
Remove unwanted copying of BlockManager in UT.
Also Delete BlockManager copy constructor & operator= so this mistake is never repeated.

Tabrizian commented:
/bot run

tensorrt-cicd commented:
[PR_Github #11834](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11834/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #11834](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11834/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #8770](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/8770/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/8770/test-results/rerun_report.html) for details.

tshmilnvidia committed:
BlockManager copy constructor fix

Remove unwanted copying of BlockManager in UT.
Also Delete BlockManager copy constructor & operator= so this mistake
is never repeated.

Signed-off-by: Tomer Shmilovich <tshmilovich@nvidia.com>

Tabrizian commented:
/bot run

tensorrt-cicd commented:
[PR_Github #11958](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11958/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #8875](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/8875/) completed with status: 'FAILURE'
